src/globus_sdk/experimental/login_flow_manager/login_flow_manager.py

```diff
@@ -18,12 +26,69 @@ def __init__(
     ):
+        """
+        :param login_client: The client to use for login flows.
+        :param request_refresh_tokens: Control whether refresh tokens will be requested.
+        :param native_prefill_named_grant: The name of a prefill in a Native App login
+            flow. This value will be ignored if the login_client is not a
+            NativeAppAuthClient.
+        """
+        if not isinstance(login_client, NativeAppAuthClient) and not isinstance(
+            login_client, ConfidentialAppAuthClient
+        ):
+            raise GlobusSDKUsageError(
+                f"{type(self).__name__} requires a NativeAppAuthClient or "
+                f"ConfidentialAppAuthClient, but got a {type(login_client).__name__}."
+            )
+
         self.login_client = login_client
         self.request_refresh_tokens = request_refresh_tokens
+        self.native_prefill_named_grant = native_prefill_named_grant
+
+    def _get_authorize_url(
+        self, auth_parameters: GlobusAuthorizationParameters, redirect_uri: str
+    ) -> str:
         """
-        :param request_refresh_tokens: Control whether refresh tokens will be requested.
+        Utility method to provide a simpler interface for subclasses to start an
+        authorization flow and get an authorization URL.
+        """
+        self._oauth2_start_flow(auth_parameters, redirect_uri)
+
+        session_required_single_domain = auth_parameters.session_required_single_domain
+        return self.login_client.oauth2_get_authorize_url(
+            session_required_identities=auth_parameters.session_required_identities,
+            session_required_single_domain=session_required_single_domain,
+            session_required_policies=auth_parameters.session_required_policies,
+            session_required_mfa=auth_parameters.session_required_mfa,
+            prompt=auth_parameters.prompt,  # type: ignore
+        )
+
+    def _oauth2_start_flow(
+        self, auth_parameters: GlobusAuthorizationParameters, redirect_uri: str
+    ) -> None:
+        """
+        Start an authorization flow with the class's login_client, returning an
+        authorization URL to direct a user to.
         """
+        login_client = self.login_client
+        requested_scopes = auth_parameters.required_scopes
+        # Native and Confidential App clients have different signatures for this method,
+        # so they must be type checked & called independently.
+        if isinstance(login_client, NativeAppAuthClient):
+            login_client.oauth2_start_flow(
+                requested_scopes,
+                redirect_uri=redirect_uri,
+                refresh_tokens=self.request_refresh_tokens,
+                prefill_named_grant=self.native_prefill_named_grant,
+            )
+        elif isinstance(login_client, ConfidentialAppAuthClient):
+            login_client.oauth2_start_flow(
+                redirect_uri,
+                requested_scopes,
```

Comment:
Nit, use parallel signatures for readability:

```suggestion
                requested_scopes,
                redirect_uri=redirect_uri,
```

Reply:
This was unfortunately done very intentionally.

Allow me to present to you the function signatures for `oauth2_start_flow` (keep in mind that these are only declared on the subclasses, not `AuthLoginClient` itself).

**ConfidentialAppAuthClient**
```python
class ConfidentialAppAuthClient(AuthLoginClient):
    ...
    def oauth2_start_flow(
        self,
        redirect_uri: str,
        requested_scopes: ScopeCollectionType | None = None,
        *,
        state: str = "_default",
        refresh_tokens: bool = False,
    ) -> GlobusAuthorizationCodeFlowManager:
    ...
```

**NativeAppAuthClient**
```python
class NativeAppAuthClient(AuthLoginClient):
    ...
    def oauth2_start_flow(
        self,
        requested_scopes: ScopeCollectionType | None = None,
        *,
        redirect_uri: str | None = None,
        state: str = "_default",
        verifier: str | None = None,
        refresh_tokens: bool = False,
        prefill_named_grant: str | None = None,
    ) -> GlobusNativeAppFlowManager:
    ...
```

Reply:
😑 [...silently backs out of the room...]